File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSSimulcastLayerDefinitionExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.media.settings.HMSSimulcastLayerDefinition

class HMSSimulcastLayerDefinitionExtension {

    companion object{
        fun toDictionary(layerDefinition: HMSSimulcastLayerDefinition): HashMap<String, Any?> {
            val hashMap = HashMap<String, Any?>()
            hashMap["hms_simulcast_layer"] = HMSSimulcastLayerExtension.getStringFromLayer(layerDefinition.layer)
            hashMap["hms_resolution"] = HMSVideoResolutionExtension.toDictionary(layerDefinition.resolution)

            return hashMap
        }
    }
}

File: ios/Classes/Models/HMSSimulcastLayerDefinitionExtension.swift
//
//  HMSSimulcastLayerDefinitionExtension.swift
//  hmssdk_flutter
//
//  Created by Govind on 06/12/22.
//

import Foundation
import HMSSDK

class HMSSimulcastLayerDefinitionExtension {
    static func toDictionary(_ layerDefinition: HMSSimulcastLayerDefinition) -> [String: Any] {
        let dict = [
            "hms_simulcast_layer": getStringFromLayer(layer: layerDefinition.layer),
            "hms_resolution": HMSVideoResolutionExtension.toDictionary(layerDefinition.resolution)
        ] as [String: Any]

        return dict
    }
    static func getStringFromLayer(layer: HMSSimulcastLayer?) -> String {
        switch layer {
            case .high:
                return "high"
            case .mid:
                return "mid"
            case .low:
                return "low"
            default:
                return ""
            }
    }

    static func getLayerFromString(layer: String) -> HMSSimulcastLayer {
        switch layer {
        case "high":
            return .high
        case "mid":
            return .mid
        case "low":
            return .low
        default:
            return .mid
        }
    }
}
